### ISSUE TITLE
Initializing hidden state of recurrent nets

### ISSUE THREAD

**Issue by ritheshkumar95** [fchollet/keras#1492]
For problems such as Machine - Translation and Visual Question Answering , It is necessary to initialize the hidden state of an RNN using the final state of another RNN. Is there a solution to this using keras? 

Currently i can only see features to initialize the weights of the RNN but not the hidden state.

**Comment by jeffiar** [comment]
Yes, I think this is possible. If you use a stateful LSTM by including `stateful=True` when creating recurrent layer, you should be able to set `model.states` to `[hidden_state_you_wish_to_use]`.

**Comment by rahul003** [comment]
Could you expand on how to set the initial state of a LSTM to a vector. LSTM being stateful means a different thing, right? For a sample, can I set the initial state of a second LSTM to the final state of a first LSTM which has processed part of the sample?

**Comment by ScottOGrady** [comment]
Im facing the same Problem:
My setup is for classification of time series. Its a multiclass classification where based on measurements and other features (num_features=26) only one of 7 num_classes should be predicted as a label at a time. I'm also using one hot encoding for pre-processing these labels for training.
I'm using a LSTM of input-shape (num_samples, backprop_length, num_features) and afterwards TimeDistributed(Dense (7,activation = 'softmax')).

My poblem is the following:
When I'm testing my LSTM by model.predict() I see, that there is a characteristic 'vibration in the output labels. After digging a little bit, i'm aware that this is the result of the backprop_length. For each new batch during prediction, the internal state is reset. Is there a simple way to deliver the last state of a previous batch as initial state to the next batch?
From my POV it does'n make sense to use the stateful=True option of keras, since it will always copy the states if index i of the previous batch to the state of index i of the next batch. In my case of classification of sequential timeseries data this doesn't make sense at all. (Considering a low sampling rate, there is no meaning why i should iitialize a state with the state that occured backprop_length states before).
Is the solution to activate return_state=True and load this as initial state to the LSTM for each batch? I'm not aware how this should work with model.predict().

I'm hoping someone can help me out with this little issue. Any and hint help is appreciated!
